### ISSUE TITLE
Add support for stages in threads for improved support handling and update thread data model.

### ISSUE THREAD

**Issue by sanchitrk** [zyghq/zyg#34]
Add support for thread stages like
- NEW - for newly created thread
- WOC - waiting on customer
- NR - needs a response from a member
- OH - on hold probably blocked by some dependency.

Deprecations

deprecate `read` flag from the thread data model, instead use stages.
deprecate `spam` flag from the thread data model, instead use stages.

Also, let us remove the `snoozed` status - consider adding it once we get feedback. For have statuses `Todo` and `Done`

**Comment by sanchitrk** [comment]
This is now fixed.
